### PR TITLE
Adding Modified Conversion Specifiers to strftime

### PR DESCRIPTION
**Pull request by abujalski**
Adding handling Modified Conversion Specifiers to strftime JavaScript glue layer implementation.

This change fixes http://posixtest.sourceforge.net/ strftime/2-1 test.

### CODE REVIEW

File: src/library.js [hunk 1]
```diff
@@ -2792,7 +2812,7 @@ LibraryManager.library = {
         return leadingNulls(Math.ceil(daysDifference/7), 2);
       },
       '%w': function(date) {
-        var day = new Date(date.tm_year+1900, date.tm_mon+1, date.tm_mday, 0, 0, 0, 0);
+        var day = new Date(date.tm_year+1900, date.tm_mon, date.tm_mday, 0, 0, 0, 0);
```
Comment (sbc100) [inline src/library.js]: The fact that this didn't change any test results makes me think we need more test coverage here.

Comment (kripken) [reply src/library.js]: I don't think tests actually ran here. @abujalski, this and your other PRs all seem to not have CI run on the commits at all. Any idea why? It's very strange...

**Comment by sbc100** [comment]
Sounds this is strange about this PR.. circleci doesn't like it much :)

File: src/library.js (hunk 1 above)
Comment (abujalski) [reply src/library.js]: Indeed, according to outputs there's something wrong with .circleci/config.yml. I've run locally latest config validator https://circleci.com/docs/2.0/local-cli/#validate-a-circleci-config and it showed that there is duplicate 'steps' section in test-browser-firefox.

Problematic lines where added in ca9520c87c long time ago, so maybe there was some update of CircleCI validator and now it is more strict. Question is why this happened only for my fork.

Commit https://github.com/abujalski/emscripten/commit/743f25f618157b0228af1a240ad97846dde9b043 removes one of them and now CI runs.

However, now the test-upstream-browser-chrome are failing. I'll update PR when I check why.

**Comment by sbc100** [comment]
Landing he circleci change separatetly: https://github.com/emscripten-core/emscripten/pull/8659.  Let see if that works.

**Comment by kripken** [comment]
I think incoming needs to be merged into here, to fix those CI issues.

File: src/library.js (hunk 1 above)
Comment (abujalski) [reply src/library.js]: @sbc100 added more tests in https://github.com/emscripten-core/emscripten/pull/8630/commits/ff33f26ed2708b215a571aebe0859c979ad8a923 to check '%w' and '%u' format specifier.

BTW. I found that according to the spec https://pubs.opengroup.org/onlinepubs/9699919799/functions/strftime.html (and gcc on Ubuntu PC) this two formats should take only `tm_wday` field.

File: tests/core/test_strftime.cpp [hunk 2]
```diff
@@ -140,6 +140,48 @@ int main() {
   size = strftime(s, 10, "%g", &tm);
   test((size == 2) && !cmp(s, "98"), "strftime test #29", s);
 
+  tm.tm_wday = 0;
+  size = strftime(s, 2, "%w", &tm);
+  test((size == 1) && !cmp(s, "0"), "strftime test #30", s);
```
Comment (sbc100) [inline tests/core/test_strftime.cpp]: These new tests don't seem to be testing the new format specifiers (e.g. `%EY`).   Maybe I'm missing something?

Comment (abujalski) [reply tests/core/test_strftime.cpp]: Actually I've focused on test to check `%w` as was pointed out in https://github.com/emscripten-core/emscripten/pull/8630#discussion_r285136190

I'll add more tests new format specifiers.

**Comment by kripken** [comment]
lgtm, thanks! Waiting to see if @sbc100 has comments.

**Pull request merged**